OfficeDevPnP.Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectPages.cs

```diff
@@ -69,6 +69,12 @@ public override void ProvisionObjects(Web web, ProvisioningTemplate template)
                     web.AddWikiPageByUrl(url);
                     web.AddLayoutToWikiPage(page.Layout, url);
                 }
+
+                if (page.WelcomePage)
+                {
+                   web.SetHomePageByServerRelativePath(url);
```

Comment:
I would just convert the serverrelative url to a folder relative url and then send that to the existing sethomepage method and not introduce a new extension.

< untested code >
var rootFolderRelativeUrl = url.Substring(web.RootFolder.ServerRelativeUrl.Length);
web.SetHomePage(rootFolderRelativeUrl);
</ untested code >

Sorry for nitpicking. I do think the rest of the PR is absolutely valid and good additions :-)